hyperspy/io_plugins/fei.py

```diff
@@ -580,13 +622,48 @@ def ser_reader(filename, objects=None, verbose=False, *args, **kwds):
         'mapping': mapping}
     return dictionary
 
-
-def get_mode(mode):
+def guess_units_from_mode(objects_dict, header, verbose=False):
```

Comment:
I think that this isn't going to work because the xmls are not stored in the order of the ser file numbers. Instead they use UUIDs to match the ser files to the xml objects. In order to match them we need to know how to generate those UUIDs.